File: src/day2/Fahrzeug.java
package day2;

public abstract class Fahrzeug {

	private String farbe;
	private int baujahr;
	
	public Fahrzeug() {
		super();
	}
	
	public Fahrzeug(String farbe, int baujahr) {
		this.farbe = farbe;
		this.baujahr = baujahr;
	}
	
	public abstract void fahren();

	public String toString() {
		return "Farbe=" + this.farbe + ", Baujahr=" + this.baujahr;
	}

	public String getFarbe() {
		return farbe;
	}

	public void setFarbe(String farbe) {
		this.farbe = farbe;
	}

	public int getBaujahr() {
		return baujahr;
	}

	public void setBaujahr(int baujahr) {
		this.baujahr = baujahr;
	}
	
}

File: lab/04-classes/fahrzeug.py

class Fahrzeug:
    def __init__(self, typ, farbe, baujahr):
        self.typ = typ
        self.farbe = farbe
        self.baujahr = baujahr

    def fahren(self):
        print(self.typ, "fährt...")

    def print(self):
        print( self.to_string() )

    def to_string(self):
        return self.typ + ": farbe = " + self.farbe + ", baujahr = " + self.baujahr

class Fahrrad(Fahrzeug):
    def __init__(self, farbe, baujahr, marke):
        super().__init__("Fahrrad", farbe, baujahr)
        self.marke = marke

    def print(self):
        print(self.to_string() + ", marke = " + self.marke)


class PKW(Fahrzeug):
    def __init__(self, farbe, baujahr, sitze):
        super().__init__("PKW", farbe, baujahr)
        self.sitze = sitze

    def print(self):
        print(self.to_string() + ", sitze = " + str(self.sitze))


if __name__ == "__main__":
    fahrzeuge = [
        Fahrrad("blau", "2015", "Scott"),
        Fahrrad("rot", "2012", "Thömues"),
        PKW("grün", "2021", "5")
    ]

    for f in fahrzeuge:
        f.print()
        f.fahren()
